Add MacOS build packaging tool (#734)
Deep-MI/FastSurfer: doc/overview/INSTALL.md

docker pull deepmi/fastsurfer:latest
```

> [Example 1](EXAMPLES.md#example-1-fastsurfer-docker) explains how to run FastSurfer (for the full pipeline you will also need a FreeSurfer .license file!) and you can find details on how to [build your own image](https://github.com/Deep-MI/FastSurfer/blob/dev/tools/Docker/README.md). 

Reviewer: ```suggestion
[Example 1](EXAMPLES.md#example-1-fastsurfer-docker) explains how to run FastSurfer (for the full pipeline you will also need a FreeSurfer .license file!) and you can find details on how to [build your own image](../../tools/Docker/README.md). 
```

```
Note, that this docker image is experimental, uses a different Python version and python packages, so results can differ from our validation results. Please do visual QC.

> ## MacOS

Reviewer: We should add a full new section on how to install the new package for macOS.
This should include where to download (include a link here), and instructions on how to install.
Also update the link in README.md to the installation instructions.
This would be the new "macOS native" installation (but really, it is not limited to M-chip systems).

Author: Currently, links are forwarding to 2.5.0-dev packages created in my repository. 
Should we package 2.4.2 release, add it to the release report and then change the links?

Continue with the example in [Example 1](EXAMPLES.md#example-1-fastsurfer-docker). 

### Package

#### 1. Requirements
FastSurfer requires pre-installed python3.10+ and bash (at least 3.2). 
You can install these via the packet manager brew.
To install brew and then python3.10, execute the following in a Terminal:

```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install python@3.10
```

### 2. FastSurfer package
From version 2.5 onward, FastSurfer ships a macOS installer package, which you can download from 
[github](https://github.com/Deep-MI/FastSurfer/releases/). 

> There are package installers for each both the Apply M-chip architecture (`arm64`) and for legacy intel chips (`x86_64`).

Reviewer 2: fix: " for each both " and fix "Apply"

> To install, drag the installer into Applications and follow the installer instructions. 

Reviewer 2: Is this true? Usually you need to double click the installer, it opens a window where you drag the App into Applications, but that is self explanatory, so double click installer should be OK. Can you confirm this?

Author: Basically, package installer is more of a common installation wizard, but it only installs apps into /Applications folder or any other folder that has been chosen as the default for the installer.
Opening installer and dragging app into applications is what dmg installers do, and they usually are used to install standalone apps, while in our case package contains both the app and the folder with the FastSurfer source code.
Please inform me if this needs to be changed. (fe. using dmg for app and separate package for source code)

Reviewer 2: I don't think it needs to be changed. But I was wondering why you drag the installer into applications as a first step? Usually installers do not get put there. Maybe the description is not complete or correct?
Also how do you follow the instructions after dragging it, I guess you start it first?

Author: Fixed in the recent commit.

After installation, you can find FastSurfer applet, its source code and freesurfer in the `/Applications` folder.

### 3. Launching FastSurfer

> To easily get a configured FastSurfer session/terminal, start the FastSurfer applet by launching it from Applications.

Reviewer 2: -> "To launch a FastSurfer terminal session, start ..."
(Suggestions don't work for some reason).

Reviewer 2: ```suggestion
To launch a configured FastSurfer terminal session, start the FastSurfer applet from Applications.
```

Continue with the example in [Example 1](EXAMPLES.md#example-1-fastsurfer-docker). 

### Native

#### 1. Dependency packages
If you do not have python3.10 or bash (at least 3.2) you can install these via the packet manager brew.

> This installs brew and then python3.10:

Reviewer: ```suggestion
### Package

#### 1. Requirements
FastSurfer requires pre-installed python3.10+ and bash (at least 3.2). 
You can install these via the packet manager brew.
To install brew and then python3.10, execute the following in a Terminal:
```

```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install python@3.10
```

### 2. FastSurfer package
Install FastSurfer package according to your type of chip architecture and follow the installer instructions.

ARM-based chips: [FastSurfer250-dev-macos-darwin_arm64](https://github.com/OtabekRintaro/FastSurfer/releases/download/2.5.0-dev/FastSurfer250-dev-macos-darwin_arm64.pkg)

> Intel chips: [FastSurfer250-dev-macos-darwin_x86_64.pkg](https://github.com/OtabekRintaro/FastSurfer/releases/download/2.5.0-dev/FastSurfer250-dev-macos-darwin_x86_64.pkg)

Reviewer: ```suggestion
From version 2.5 onward, FastSurfer ships a macOS installer package, which you can download from 
[github](https://github.com/Deep-MI/FastSurfer/releases/). 
There are package installers for each both the Apply M-chip architecture (`arm64`) and for legacy intel chips (`x86_64`).
To install, drag the installer into Applications and follow the installer instructions. 
```

After installation, you can find FastSurfer applet, its source code and freesurfer in the `/Applications` folder.

> ### 3 Launching FastSurfer

Reviewer: ```suggestion
### 3. Launching FastSurfer
```

Launching FastSurfer applet will open terminal and setup environment for FastSurfer.

> Once it is set, you can run the full FastSurfer pipeline only by calling ```run_fastsurfer.sh ....``` with the appropriate command line flags, see the [commandline documentation](../../README.md#usage).

Reviewer: ```suggestion
To easily get a configured FastSurfer session/terminal, start the FastSurfer applet by launching it from Applications.

Once the terminal opens, it is already configured and you can easily run the full FastSurfer pipeline by typing and executing `run_fastsurfer.sh <fastsurfer-flags>`, where you replace `<fastsurfer-flags>` with the appropriate [commandline flags of FastSurfer](../../README.md#usage).
```

Continue with the example in [Example 1](EXAMPLES.md#example-1-fastsurfer-docker). 

### Package

#### 1. Requirements
FastSurfer requires pre-installed python3.10+ and bash (at least 3.2). 
You can install these via the packet manager brew.
To install brew and then python3.10, execute the following in a Terminal:

```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install python@3.10
```

### 2. FastSurfer package
From version 2.5 onward, FastSurfer ships a macOS installer package, which you can download from 
[github](https://github.com/Deep-MI/FastSurfer/releases/). 

> There are package installers for each both the Apply M-chip architecture (`arm64`) and for legacy intel chips (`x86_64`).

Reviewer 2: ```suggestion
There are package installers for both the Apple M-chip architecture (`arm64`) and for legacy Intel chips (`x86_64`).
```

To install, drag the installer into Applications and follow the installer instructions. 

> After installation, you can find FastSurfer applet, its source code and freesurfer in the `/Applications` folder.

Reviewer 2: ```suggestion
After installation, you can find the FastSurfer applet, its source code and FreeSurfer in the `/Applications` folder.
```

> After installation, you can find the FastSurfer applet, its source code and FreeSurfer in the `/Applications` folder.

Reviewer 2: Is it true that FreeSurfer is also fully installed ? Or only the pruned version ? 
Depending on the answer this may need further discussion.

Reviewer: We only install the pruned freesurfer (using the same pruning script as for the docker)

Reviewer 2: ```suggestion
After installation, you can find the FastSurfer applet, its source code, and selected FreeSurfer executables in the `/Applications` folder.
```
